pkg/queue/workers/metrics.go

```diff
@@ -0,0 +1,174 @@
+package workers
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// largest bucket is 83 min
+var (
+	durationSBuckets  = []float64{1, 10, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000}
+	durationMsBuckets = []float64{10, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000}
+	queueMetricLabels = []string{"queue"}
+	taskMetricLabels  = []string{"queue", "type"}
+)
+
+// TaskQueueMetrics provides access to the prometheus metric objects for the task queue
+var TaskQueueMetrics = struct {
+	Labels             []string
+	TaskDuration       *prometheus.HistogramVec
+	TaskWaiting        *prometheus.HistogramVec
+	DequeueDuration    prometheus.Histogram
+	WorkerGauge        prometheus.Gauge
+	WorkerWaiting      prometheus.Counter
+	WorkerWorking      *prometheus.CounterVec
+	WorkerWorkingGauge *prometheus.GaugeVec
+	WorkerTask         *prometheus.CounterVec
+	WorkerErrors       *prometheus.CounterVec
+	QueueErrors        prometheus.Counter
+}{
+	Labels: queueMetricLabels,
+	TaskDuration: promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "hub",
```

Comment:
same about these metrics, need to be parameterized and stored inside an instance.

Reply:
done